### ISSUE TITLE
Error compiling on PHP 7.0 RC1 on OSX 10.10.4

### ISSUE THREAD

**Issue by andrewvy** [krakjoe/pthreads#476]
```console
/Users/andrewvy/Sites/pthreads/src/store.c:219:5: error: conflicting types for 'pthreads_store_count'
int pthreads_store_count(zval *object, zend_long *count) {
    ^
/Users/andrewvy/Sites/pthreads/src/store.h:47:5: note: previous declaration is here
int pthreads_store_count(zval *object, long *count);
    ^
/Users/andrewvy/Sites/pthreads/src/store.c:258:5: error: conflicting types for 'pthreads_store_chunk'
int pthreads_store_chunk(zval *object, zend_long size, zend_bool preserve, zval *chunk) {
    ^
/Users/andrewvy/Sites/pthreads/src/store.h:45:5: note: previous declaration is here
int pthreads_store_chunk(zval *object, long size, zend_bool preserve, zval *chunk);
    ^
/Users/andrewvy/Sites/pthreads/src/store.c:519:20: warning: implicit declaration of function 'pthreads_globals_object_validate' is invalid in C99 [-Wimplicit-function-declaration]
```

Should I try on current php-src master?

**Comment by krakjoe** [comment]
Should be okay now ?

**Comment by andrewvy** [comment]
:+1: Compiles perfectly now. I'm running into segfaults when running any example files though, I'll get back with a new issue if I find anything.

**Comment by krakjoe** [comment]
Which ones ?

**Comment by andrewvy** [comment]
Ah yeah. I've ran the tests and all of them fail (the ones that use pthread code.). And all of the examples seem to result in a segfault as well, sadly. :cry: 

```
Running 63 tests
FAIL [ 1/63] Test anonymous classes (bound inherited class)[anon-bound-inherited.phpt]
FAIL [ 2/63] Test anonymous classes (unbound inherited class)[anon-unbound-inherited.phpt]
FAIL [ 3/63] Test pthreads connections[binding.phpt]
FAIL [ 4/63] Testing extended object functionality, magic methods bug #34[callstatic-etc.phpt]
PASS [ 5/63] Check chunking[chunk.phpt]
FAIL [ 6/63] Test class defaults[class-defaults.phpt]
FAIL [ 7/63] Testing closure members[closures-as-members.phpt]
FAIL [ 8/63] Test NULLing ressources in arrays[complex-statics-set-null.phpt]
FAIL [ 9/63] Test complex statics bug #32[complex-statics.phpt]
FAIL [10/63] Test constants (bug 19)[constants.phpt]
FAIL [11/63] Testing sane handling of resources and objects bug #39[corrupted-resources.phpt]
PASS [12/63] Testing member count[counting-members.phpt]
FAIL [13/63] Test managing refcount in userland (super advanced voodoo shit, leave it alone)[crazy-refcounting-stuff.phpt]
FAIL [14/63] Test doc comments are copied properly[doc-comments.phpt]
FAIL [15/63] Test function table inheritance[exception-handler.phpt]
FAIL [16/63] Testing foreign private member access, magic methods bug #32[foreign-private-members.phpt]
FAIL [17/63] Test function table inheritance[functions.phpt]
FAIL [18/63] Test global constants[global-constants.phpt]
FAIL [19/63] Test objects that have gone away[gone.phpt]
FAIL [20/63] Test graceful fatalities[graceful-fatalities.phpt]
FAIL [21/63] Test include/require functions as expected[included-files.phpt]
FAIL [22/63] Test INI inheritance (bug 20)[ini.phpt]
FAIL [23/63] Test interface functionality[interface.phpt]
FAIL [24/63] Testing interfaces inheritance (#124)[interfaces.phpt]
PASS [25/63] Test isset and empty are compliant with standard objects (fix bug #347)[isset-empty-compat.phpt]
PASS [26/63] Testing isset unset[isset-unset.phpt]
FAIL [27/63] Test kill[killed-info.phpt]
FAIL [28/63] Test legacy constructor (issue #336)[legacy-ctor.phpt]
PASS [29/63] Test magic __get and __set[magic-methods-get-set.phpt]
PASS [30/63] Test magic __isset and __unset[magic-methods-isset-unset.phpt]
PASS [31/63] Test merge non-threaded[merge-non-threaded.phpt]
PASS [32/63] Testing merging members[merging-members.phpt]
PASS [33/63] Testing merging members (long keys)[merging-range.phpt]
PASS [34/63] Test members (typeof object) with no other references[norefs.phpt]
FAIL [35/63] Test reading object properties without debug info[normal-reads.phpt]
PASS [36/63] Testing normalizing members[normalize-members.phpt]
PASS [37/63] Null member crash[null-member-crash.phpt]
FAIL [38/63] Test access to user defined methods in the object context[oomethods.phpt]
FAIL [39/63] Test pool defaults[pools-default.phpt]
FAIL [40/63] Test pooling[pools.phpt]
FAIL [41/63] Test sane handling of resources from foreign contexts[resource-keeping.phpt]
FAIL [42/63] Test return types[return-types.phpt]
PASS [43/63] Test runtime extension[runtime-inheritance.phpt]
FAIL [44/63] Test selective inheritance[selective-inheritance.phpt]
PASS [45/63] Test shift and pop[shift-pop.phpt]
FAIL [46/63] Test shutdown handlers #204[shutdown-handler.phpt]
FAIL [47/63] Test stateful fatalities[stateful-fatalities.phpt]
FAIL [48/63] Test static:: regression[static-resolution.phpt]
FAIL [49/63] Test access to static methods from within user threads[staticmethods.phpt]
FAIL [50/63] Test statics aren't getting fucked with when starting new threads (many bugs)[statics-thorn-in-other-side.phpt]
FAIL [51/63] Test statics (bug 19)[statics.phpt]
FAIL [52/63] Test synchronized blocks[synchronization.phpt]
PASS [53/63] Testing thorn in my side hasn't returned[thorn-in-side.phpt]
FAIL [54/63] Test basic threading[thread.phpt]
PASS [55/63] Test extending abstract[threaded-extend-abstract.phpt]
FAIL [56/63] Test trait aliases[trait-aliases.phpt]
FAIL [57/63] Test unset defaults [moot: default values are ignored][unset-defaults.phpt]
FAIL [58/63] Test wait/notify[wait.phpt]
FAIL [59/63] Test waiting timeouts[waiting-timeouts.phpt]
FAIL [60/63] Test pthreads workers rules (collect)[workers-no-collect.phpt]
FAIL [61/63] Test pthreads workers rules (join)[workers-no-join.phpt]
FAIL [62/63] Test pthreads workers rules (stack)[workers-no-stack.phpt]
FAIL [63/63] Test pthreads workers rules (start)[workers-no-start.phpt]
wrote log to "/Users/andrewvy/Sites/pthreads/tests/run-tests.log"
TOTAL TIME: 00:24
16 PASSED TESTS
0 SKIPPED TESTS
47 FAILED TESTS:
/Users/andrewvy/Sites/pthreads/tests/anon-bound-inherited.phpt
/Users/andrewvy/Sites/pthreads/tests/anon-unbound-inherited.phpt
/Users/andrewvy/Sites/pthreads/tests/binding.phpt
/Users/andrewvy/Sites/pthreads/tests/callstatic-etc.phpt
/Users/andrewvy/Sites/pthreads/tests/class-defaults.phpt
/Users/andrewvy/Sites/pthreads/tests/closures-as-members.phpt
/Users/andrewvy/Sites/pthreads/tests/complex-statics-set-null.phpt
/Users/andrewvy/Sites/pthreads/tests/complex-statics.phpt
/Users/andrewvy/Sites/pthreads/tests/constants.phpt
/Users/andrewvy/Sites/pthreads/tests/corrupted-resources.phpt
/Users/andrewvy/Sites/pthreads/tests/crazy-refcounting-stuff.phpt
/Users/andrewvy/Sites/pthreads/tests/doc-comments.phpt
/Users/andrewvy/Sites/pthreads/tests/exception-handler.phpt
/Users/andrewvy/Sites/pthreads/tests/foreign-private-members.phpt
/Users/andrewvy/Sites/pthreads/tests/functions.phpt
/Users/andrewvy/Sites/pthreads/tests/global-constants.phpt
/Users/andrewvy/Sites/pthreads/tests/gone.phpt
/Users/andrewvy/Sites/pthreads/tests/graceful-fatalities.phpt
/Users/andrewvy/Sites/pthreads/tests/included-files.phpt
/Users/andrewvy/Sites/pthreads/tests/ini.phpt
/Users/andrewvy/Sites/pthreads/tests/interface.phpt
/Users/andrewvy/Sites/pthreads/tests/interfaces.phpt
/Users/andrewvy/Sites/pthreads/tests/killed-info.phpt
/Users/andrewvy/Sites/pthreads/tests/legacy-ctor.phpt
/Users/andrewvy/Sites/pthreads/tests/normal-reads.phpt
/Users/andrewvy/Sites/pthreads/tests/oomethods.phpt
/Users/andrewvy/Sites/pthreads/tests/pools-default.phpt
/Users/andrewvy/Sites/pthreads/tests/pools.phpt
/Users/andrewvy/Sites/pthreads/tests/resource-keeping.phpt
/Users/andrewvy/Sites/pthreads/tests/return-types.phpt
/Users/andrewvy/Sites/pthreads/tests/selective-inheritance.phpt
/Users/andrewvy/Sites/pthreads/tests/shutdown-handler.phpt
/Users/andrewvy/Sites/pthreads/tests/stateful-fatalities.phpt
/Users/andrewvy/Sites/pthreads/tests/static-resolution.phpt
/Users/andrewvy/Sites/pthreads/tests/staticmethods.phpt
/Users/andrewvy/Sites/pthreads/tests/statics-thorn-in-other-side.phpt
/Users/andrewvy/Sites/pthreads/tests/statics.phpt
/Users/andrewvy/Sites/pthreads/tests/synchronization.phpt
/Users/andrewvy/Sites/pthreads/tests/thread.phpt
/Users/andrewvy/Sites/pthreads/tests/trait-aliases.phpt
/Users/andrewvy/Sites/pthreads/tests/unset-defaults.phpt
/Users/andrewvy/Sites/pthreads/tests/wait.phpt
/Users/andrewvy/Sites/pthreads/tests/waiting-timeouts.phpt
/Users/andrewvy/Sites/pthreads/tests/workers-no-collect.phpt
/Users/andrewvy/Sites/pthreads/tests/workers-no-join.phpt
/Users/andrewvy/Sites/pthreads/tests/workers-no-stack.phpt
/Users/andrewvy/Sites/pthreads/tests/workers-no-start.phpt
Some tests failed
```

I'll put together some valgrind logs and any extra info in a new github issue for you.

**Comment by krakjoe** [comment]
Ok cool, I'll get a vm together today too ... thanks for taking the time ;)

**Comment by krakjoe** [comment]
```
=====================================================================
PHP         : /Users/krakjoe/php-7.0.0RC1/../php7/bin/php 
PHP_SAPI    : cli
PHP_VERSION : 7.0.0RC1
ZEND_VERSION: 3.0.0-dev
PHP_OS      : Darwin - Darwin galaxycore.de 13.1.0 Darwin Kernel Version 13.1.0: Thu Jan 16 19:40:37 PST 2014; root:xnu-2422.90.20~2/RELEASE_X86_64 x86_64
INI actual  : /Users/krakjoe/pthreads/tmp-php.ini
More .INIs  :   
CWD         : /Users/krakjoe/pthreads
Extra dirs  : 
VALGRIND    : Not used
=====================================================================
TIME START 2015-08-26 10:37:49
=====================================================================
PASS Test anonymous classes (bound inherited class) [tests/anon-bound-inherited.phpt] 
PASS Test anonymous classes (unbound inherited class) [tests/anon-unbound-inherited.phpt] 
PASS Test pthreads connections [tests/binding.phpt] 
PASS Testing extended object functionality, magic methods bug #34 [tests/callstatic-etc.phpt] 
PASS Check chunking [tests/chunk.phpt] 
PASS Test class defaults [tests/class-defaults.phpt] 
PASS Testing closure members [tests/closures-as-members.phpt] 
PASS Test NULLing ressources in arrays [tests/complex-statics-set-null.phpt] 
PASS Test complex statics bug #32 [tests/complex-statics.phpt] 
PASS Test constants (bug 19) [tests/constants.phpt] 
PASS Testing sane handling of resources and objects bug #39 [tests/corrupted-resources.phpt] 
PASS Testing member count [tests/counting-members.phpt] 
PASS Test managing refcount in userland (super advanced voodoo shit, leave it alone) [tests/crazy-refcounting-stuff.phpt] 
PASS Test doc comments are copied properly [tests/doc-comments.phpt] 
PASS Test function table inheritance [tests/exception-handler.phpt] 
PASS Testing foreign private member access, magic methods bug #32 [tests/foreign-private-members.phpt] 
PASS Test function table inheritance [tests/functions.phpt] 
PASS Test global constants [tests/global-constants.phpt] 
PASS Test objects that have gone away [tests/gone.phpt] 
PASS Test graceful fatalities [tests/graceful-fatalities.phpt] 
PASS Test include/require functions as expected [tests/included-files.phpt] 
PASS Test INI inheritance (bug 20) [tests/ini.phpt] 
PASS Test interface functionality [tests/interface.phpt] 
PASS Testing interfaces inheritance (#124) [tests/interfaces.phpt] 
PASS Test isset and empty are compliant with standard objects (fix bug #347) [tests/isset-empty-compat.phpt] 
PASS Testing isset unset [tests/isset-unset.phpt] 
PASS Test kill [tests/killed-info.phpt] 
PASS Test legacy constructor (issue #336) [tests/legacy-ctor.phpt] 
PASS Test magic __get and __set [tests/magic-methods-get-set.phpt] 
PASS Test magic __isset and __unset [tests/magic-methods-isset-unset.phpt] 
PASS Test merge non-threaded [tests/merge-non-threaded.phpt] 
PASS Testing merging members [tests/merging-members.phpt] 
PASS Testing merging members (long keys) [tests/merging-range.phpt] 
PASS Test members (typeof object) with no other references [tests/norefs.phpt] 
PASS Test reading object properties without debug info [tests/normal-reads.phpt] 
PASS Testing normalizing members [tests/normalize-members.phpt] 
PASS Null member crash [tests/null-member-crash.phpt] 
PASS Test access to user defined methods in the object context [tests/oomethods.phpt] 
PASS Test pool defaults [tests/pools-default.phpt] 
PASS Test pooling [tests/pools.phpt] 
PASS Test sane handling of resources from foreign contexts [tests/resource-keeping.phpt] 
PASS Test return types [tests/return-types.phpt] 
PASS Test runtime extension [tests/runtime-inheritance.phpt] 
PASS Test selective inheritance [tests/selective-inheritance.phpt] 
PASS Test shift and pop [tests/shift-pop.phpt] 
PASS Test shutdown handlers #204 [tests/shutdown-handler.phpt] 
PASS Test stateful fatalities [tests/stateful-fatalities.phpt] 
PASS Test static:: regression [tests/static-resolution.phpt] 
PASS Test access to static methods from within user threads [tests/staticmethods.phpt] 
PASS Test statics aren't getting fucked with when starting new threads (many bugs) [tests/statics-thorn-in-other-side.phpt] 
PASS Test statics (bug 19) [tests/statics.phpt] 
PASS Test synchronized blocks [tests/synchronization.phpt] 
PASS Testing thorn in my side hasn't returned [tests/thorn-in-side.phpt] 
PASS Test basic threading [tests/thread.phpt] 
PASS Test extending abstract [tests/threaded-extend-abstract.phpt] 
PASS Test trait aliases [tests/trait-aliases.phpt] 
PASS Test unset defaults [moot: default values are ignored] [tests/unset-defaults.phpt] 
PASS Test wait/notify [tests/wait.phpt] 
FAIL Test waiting timeouts [tests/waiting-timeouts.phpt] 
PASS Test pthreads workers rules (collect) [tests/workers-no-collect.phpt] 
PASS Test pthreads workers rules (join) [tests/workers-no-join.phpt] 
PASS Test pthreads workers rules (stack) [tests/workers-no-stack.phpt] 
PASS Test pthreads workers rules (start) [tests/workers-no-start.phpt] 
=====================================================================
TIME END 2015-08-26 10:38:01

=====================================================================
TEST RESULT SUMMARY
---------------------------------------------------------------------
Exts skipped    :    0
Exts tested     :   27
---------------------------------------------------------------------

Number of tests :   63                63
Tests skipped   :    0 (  0.0%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :    1 (  1.6%) (  1.6%)
Expected fail   :    0 (  0.0%) (  0.0%)
Tests passed    :   62 ( 98.4%) ( 98.4%)
---------------------------------------------------------------------
Time taken      :   12 seconds
=====================================================================

=====================================================================
FAILED TEST SUMMARY
---------------------------------------------------------------------
Test waiting timeouts [tests/waiting-timeouts.phpt]
=====================================================================
```

More info needed, can't reproduce :(

**Comment by andrewvy** [comment]
This is good! At least it's possible to be working. Can you give your installation instructions?

I used https://github.com/Homebrew/homebrew-php `php70` formula with the ZTS enabled.
I cloned `pthreads` and ran `pear build`.
I also made sure the module was loaded with `php -m`.

I should just try to do a clean build from src, maybe that will help.

**Comment by krakjoe** [comment]
I downloaded RC one source, configured with --disable-all and --enable-maintainer-zts and then built pthreads and got that result ... try disable-all configure switch when building php, then if that works, enable each extension until you know which one is crashing it ...

**Comment by andrewvy** [comment]
It works!

```
=====================================================================
PHP         : /usr/local/bin/php
PHP_SAPI    : cli
PHP_VERSION : 7.0.0RC1
ZEND_VERSION: 3.0.0-dev
PHP_OS      : Darwin - Darwin Project-Kittens.local 14.4.0 Darwin Kernel Version 14.4.0: Thu May 28 11:35:04 PDT 2015; root:xnu-2782.30.5~1/RELEASE_X86_64 x86_64
INI actual  : /Users/andrewvy/Sites/pthreads/tmp-php.ini
More .INIs  :
CWD         : /Users/andrewvy/Sites/pthreads
Extra dirs  :
VALGRIND    : Not used
=====================================================================
TIME START 2015-08-27 16:50:49
=====================================================================
PASS Test anonymous classes (bound inherited class) [tests/anon-bound-inherited.phpt]
PASS Test anonymous classes (unbound inherited class) [tests/anon-unbound-inherited.phpt]
PASS Test pthreads connections [tests/binding.phpt]
PASS Testing extended object functionality, magic methods bug #34 [tests/callstatic-etc.phpt]
PASS Check chunking [tests/chunk.phpt]
PASS Test class defaults [tests/class-defaults.phpt]
PASS Testing closure members [tests/closures-as-members.phpt]
PASS Test NULLing ressources in arrays [tests/complex-statics-set-null.phpt]
PASS Test complex statics bug #32 [tests/complex-statics.phpt]
PASS Test constants (bug 19) [tests/constants.phpt]
PASS Testing sane handling of resources and objects bug #39 [tests/corrupted-resources.phpt]
PASS Testing member count [tests/counting-members.phpt]
PASS Test managing refcount in userland (super advanced voodoo shit, leave it alone) [tests/crazy-refcounting-stuff.phpt]
PASS Test doc comments are copied properly [tests/doc-comments.phpt]
PASS Test function table inheritance [tests/exception-handler.phpt]
PASS Testing foreign private member access, magic methods bug #32 [tests/foreign-private-members.phpt]
PASS Test function table inheritance [tests/functions.phpt]
PASS Test global constants [tests/global-constants.phpt]
PASS Test objects that have gone away [tests/gone.phpt]
PASS Test graceful fatalities [tests/graceful-fatalities.phpt]
PASS Test include/require functions as expected [tests/included-files.phpt]
PASS Test INI inheritance (bug 20) [tests/ini.phpt]
PASS Test interface functionality [tests/interface.phpt]
PASS Testing interfaces inheritance (#124) [tests/interfaces.phpt]
PASS Test isset and empty are compliant with standard objects (fix bug #347) [tests/isset-empty-compat.phpt]
PASS Testing isset unset [tests/isset-unset.phpt]
FAIL Test kill [tests/killed-info.phpt]
PASS Test legacy constructor (issue #336) [tests/legacy-ctor.phpt]
PASS Test magic __get and __set [tests/magic-methods-get-set.phpt]
PASS Test magic __isset and __unset [tests/magic-methods-isset-unset.phpt]
PASS Test merge non-threaded [tests/merge-non-threaded.phpt]
PASS Testing merging members [tests/merging-members.phpt]
PASS Testing merging members (long keys) [tests/merging-range.phpt]
PASS Test members (typeof object) with no other references [tests/norefs.phpt]
PASS Test reading object properties without debug info [tests/normal-reads.phpt]
PASS Testing normalizing members [tests/normalize-members.phpt]
PASS Null member crash [tests/null-member-crash.phpt]
PASS Test access to user defined methods in the object context [tests/oomethods.phpt]
PASS Test pool defaults [tests/pools-default.phpt]
PASS Test pooling [tests/pools.phpt]
PASS Test sane handling of resources from foreign contexts [tests/resource-keeping.phpt]
PASS Test return types [tests/return-types.phpt]
PASS Test runtime extension [tests/runtime-inheritance.phpt]
PASS Test selective inheritance [tests/selective-inheritance.phpt]
PASS Test shift and pop [tests/shift-pop.phpt]
PASS Test shutdown handlers #204 [tests/shutdown-handler.phpt]
PASS Test stateful fatalities [tests/stateful-fatalities.phpt]
PASS Test static:: regression [tests/static-resolution.phpt]
PASS Test access to static methods from within user threads [tests/staticmethods.phpt]
PASS Test statics aren't getting fucked with when starting new threads (many bugs) [tests/statics-thorn-in-other-side.phpt]
PASS Test statics (bug 19) [tests/statics.phpt]
PASS Test synchronized blocks [tests/synchronization.phpt]
PASS Testing thorn in my side hasn't returned [tests/thorn-in-side.phpt]
PASS Test basic threading [tests/thread.phpt]
PASS Test extending abstract [tests/threaded-extend-abstract.phpt]
PASS Test trait aliases [tests/trait-aliases.phpt]
PASS Test unset defaults [moot: default values are ignored] [tests/unset-defaults.phpt]
PASS Test wait/notify [tests/wait.phpt]
FAIL Test waiting timeouts [tests/waiting-timeouts.phpt]
PASS Test pthreads workers rules (collect) [tests/workers-no-collect.phpt]
PASS Test pthreads workers rules (join) [tests/workers-no-join.phpt]
PASS Test pthreads workers rules (stack) [tests/workers-no-stack.phpt]
PASS Test pthreads workers rules (start) [tests/workers-no-start.phpt]
=====================================================================
TIME END 2015-08-27 16:51:53

=====================================================================
TEST RESULT SUMMARY
---------------------------------------------------------------------
Exts skipped    :    0
Exts tested     :   26
---------------------------------------------------------------------

Number of tests :   63                63
Tests skipped   :    0 (  0.0%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :    2 (  3.2%) (  3.2%)
Expected fail   :    0 (  0.0%) (  0.0%)
Tests passed    :   61 ( 96.8%) ( 96.8%)
---------------------------------------------------------------------
Time taken      :   64 seconds
=====================================================================

=====================================================================
FAILED TEST SUMMARY
---------------------------------------------------------------------
Test kill [tests/killed-info.phpt]
Test waiting timeouts [tests/waiting-timeouts.phpt]
=====================================================================
```

:) Thanks so much for the help! Looks like it's working, less a few oddities with those two tests.

**Comment by krakjoe** [comment]
I've expanded the description on that waiting-timeouts test, you can mostly ignore that ... I'll look and see if I can get the killed-info one to fail, but it's probably the same issue (that I don't think I can do anything about)

Glad it's working for ya ;)
